makefile

```diff
@@ -357,10 +357,41 @@ CPPCHECK_IGNORE_FLAGS = $(patsubst %,-i%, $(CPPCHECK_IGNORE))
 cppcheck:
 	cppcheck $(CPPCHECK_FLAGS) $(DEFINES) $(INC) $(CPPCHECK_DIRS) $(CPPCHECK_IGNORE_FLAGS)
 
+################################################################################
+# gcov targets
+################################################################################
+
 gen-coverage:
 	lcov --capture --directory ./emulator/obj/ --output-file ./coverage.info
 	genhtml ./coverage.info --output-directory ./coverage
 	firefox ./coverage/index.html &
 
+################################################################################
+# misc targets
+################################################################################
+
+PROPERTIES_FILE=./.vscode/c_cpp_properties.json
+
+gen-vs-properties:
+	@echo "{"                                                                                      >> $(PROPERTIES_FILE)
+	@echo "    \"configurations\": ["                                                              >> $(PROPERTIES_FILE)
+	@echo "        {"                                                                              >> $(PROPERTIES_FILE)
+	@echo "            \"name\": \"esp32s2\","                                                     >> $(PROPERTIES_FILE)
+	@echo "            \"includePath\": ["                                                         >> $(PROPERTIES_FILE)
+	@echo "                \"\$${workspaceFolder}/main/**\","                                      >> $(PROPERTIES_FILE)
+	@echo "                \"\$${workspaceFolder}/components/**\","                                >> $(PROPERTIES_FILE)
+	@echo "                \"${IDF_PATH}/components/**\""                                          >> $(PROPERTIES_FILE)
+	@echo "            ],"                                                                         >> $(PROPERTIES_FILE)
+	@echo "            \"defines\": [],"                                                           >> $(PROPERTIES_FILE)
+	@echo "            \"compilerPath\": \"$(shell which xtensa-esp32s2-elf-gcc)\","               >> $(PROPERTIES_FILE)
```

Comment:
Output is still doing some oddities with UNIX-style paths on Windows:
``` json
{
    "configurations": [
        {
            "name": "esp32s2",
            "includePath": [
                "${workspaceFolder}/main/**",
                "${workspaceFolder}/components/**",
                "C:\Users\Bryce\esp\esp-idf/components/**"
            ],
            "defines": [],
            "compilerPath": "/c/Users/Bryce/.espressif/tools/xtensa-esp32s2-elf/esp-12.2.0_20230208/xtensa-esp32s2-elf/bin/xtensa-esp32s2-elf-gcc",
            "compileCommands": "${workspaceFolder}/build/compile_commands.json",
            "cStandard": "c17",
            "cppStandard": "c++17",
            "intelliSenseMode": "gcc-x86"
        }
    ],
    "version": 4
}
```

Do we have the ability to replace this call with something that puts out Windows paths?